code-challenges.test.js

```diff
@@ -54,3 +200,68 @@ const hand3 = [5, 5, 5, 5, 4]
 
 
 // b) Create the function that makes the test pass.
+const poker = (array) => {
+    if(hand2.indexOf(4 % 2 === 0)) {
+        return false
+    }
+    else if(hand3.indexOf(4 % 2 === 0)) {
+        return false
+    }
+    else if(hand1.indexOf(4 % 2 === 1)) ;{
+        return true
+    }
+}
+   
+
+
+    
+//---------------------------------------QUESTION 3 NOTES---------------------------------------
+// first error I got 
+
+//● poker › takes in an array of 5 numbers and determines whether or not the array is a “full house
+
+// expect(received).toEqual(expected) // deep equality
+
+// Expected: true
+// Received: "false"
+
+//   168 |         const hand2 = [5, 5, 3, 3, 4]
+//   169 |         const hand3 = [5, 5, 5, 5, 4]
+// > 170 |         expect(poker(hand1)).toEqual(true)
+//       |                              ^
+//   171 |         expect(poker(hand2)).toEqual(false)
+//   172 |         expect(poker(hand2)).toEqual(false)
+//   173 |
+
+//   at Object.toEqual (code-challenges.test.js:170:30)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       3 failed, 3 total
+// Snapshots:   0 total
+// Time:        0.21 s
+//------------------------------------------------------------------------------------
+// So I changed the hard coded the parameter which mixed up some of the variables and got this error:
+
+//● poker › takes in an array of 5 numbers and determines whether or not the array is a “full house
+
+// TypeError: Cannot read properties of undefined (reading 'indexOf')
+
+// 187 | // b) Create the function that makes the test pass.
+// 188 | const poker = (hand1, hand2, hand3) => {
+// > 189 |     if(hand2.indexOf(4 % 2 === 0)) {
+//     |              ^
+// 190 |         return "false"
+// 191 |     }
+// 192 |     else if(hand3.indexOf(4 % 2 === 0)) {
+
+// at indexOf (code-challenges.test.js:189:14)
+// at Object.poker (code-challenges.test.js:170:16)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       3 failed, 3 total
+// Snapshots:   0 total
+// Time:        0.212 s, estimated 1 s
+// Ran all test suites.
+//------------------------------------------------------------------------------------
+// so then i put the array back as a parameter and got the same error as the first one.
+// This is the first test that I felt like I'm just right there!!!! I just dont know what to correct, it looks corret to me?
```

Comment:
Again, great job with the effort.  This is a tough one.  Your tests look good.  Your function needs a different approach. I'd be happy to set aside some time to work through this with you.  Just let me know.  :)